docs/design/async-checkpoint.md

```diff
@@ -0,0 +1,176 @@
+# Fastly Save and Load Checkpoint to Resume Training
+
+The design to introduce how to speed up saving and loading the checkpoint
+to resume training.
+
+## Background
+
+It is very difficult to keep training a fundational model during a lone time
+without any interruption. Hareware failure, OS failure or network breakdwon
+may oftern happen during the training. Users usually periodically checkpoint
+the model and optimizer states to the storage to resume the training
+after an interruption. Now, It takes a few minutes to use  `torch.save` to
+checkpoint a large fundational model. Frequent checkpointing may waste
+many hours of the accelerator (GPU, NPU, TPU).
+Because, the training must stop until the checkpointing is completed.
+On the other hand, too long checkpionting period will waste many training steps
+when resuming the training from the latest checkpoint file. Shortening the checkpinting
```

Comment:
Here do you mean the large size of checkpoints require long time period to resume?

Reply:
It is not. For example, we save a checkpoint per 24h and the training fails at 23h. The training resumes the model from the 23 hours old checkpoint. The 23 hours are wasted during the latest checkpointing and the interruption.

Reply:
Ok. I think it should be more clear like `On the other hand, infrequent checkpoints could reduce the overhead but take the risk of wasting more training steps when ...`